components/core/src/clp/streaming_compression/lzma/Compressor.cpp

```diff
@@ -0,0 +1,259 @@
+#include "Compressor.hpp"
+
+#include <array>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+
+#include <lzma.h>
+#include <spdlog/spdlog.h>
+
+#include "../../ErrorCode.hpp"
+#include "../../FileWriter.hpp"
+#include "../../TraceableException.hpp"
+#include "../../type_utils.hpp"
+#include "Constants.hpp"
+
+namespace {
+using clp::streaming_compression::lzma::Compressor;
+
+auto is_flush_action(lzma_action action) -> bool {
+    return LZMA_SYNC_FLUSH == action || LZMA_FULL_FLUSH == action || LZMA_FULL_BARRIER == action
+           || LZMA_FINISH == action;
+}
+
+/**
+ * Initialize the LZMA compression stream
+ * @param strm A pre-allocated `lzma_stream` object
+ * @param compression_level
+ * @param dict_size Dictionary size that specifies how many bytes of the
+ *                  recently processed uncompressed data to keep in the memory
+ */
+auto init_lzma_encoder(lzma_stream* strm, int compression_level, size_t dict_size) -> void {
+    lzma_options_lzma options;
+    if (0 != lzma_lzma_preset(&options, compression_level)) {
+        SPDLOG_ERROR("Failed to initialize LZMA options' compression level.");
+        throw Compressor::OperationFailed(clp::ErrorCode_BadParam, __FILENAME__, __LINE__);
+    }
+    options.dict_size = dict_size;
+    std::array<lzma_filter, 2> filters{{
+            {.id = LZMA_FILTER_LZMA2, .options = &options},
+            {.id = LZMA_VLI_UNKNOWN, .options = nullptr},
+    }};
+
+    // Initialize the encoder using a preset. Set the integrity to check
+    // to CRC64, which is the default in the xz command line tool. If
+    // the .xz file needs to be decompressed with XZ Embedded, use
+    // LZMA_CHECK_CRC32 instead.
+    auto const rc{lzma_stream_encoder(strm, filters.data(), LZMA_CHECK_CRC64)};
+
+    if (LZMA_OK == rc) {
+        return;
+    }
+
+    // Something went wrong. The possible errors are documented in
```

Comment:
I believe that we have the coding rule to make each line of comment to be close to 100 chars as possible.
Ditto for other comments.
@kirkrodrigues can you confirm?

Reply:
Yes: https://www.notion.so/yscope/WIP-Coding-Guidelines-9a308b847a5343958ba3cb97a850be66?pvs=4#4663c71574e846ba83a55c7b95184fa9